subprojects/packagefiles/cmark-gfm/meson.build

```diff
@@ -0,0 +1,123 @@
+project(
+    'cmark-gfm',
+    'c',
+    version: '0.29.0.gfm.6',
+    default_options: ['warning_level=3'],
+    meson_version: '>=0.49.0',
+    license : 'BSD-2-Clause',
+)
+
+cc = meson.get_compiler('c')
+has_stdbool_header = cc.has_header('stdbool.h')
+has_builtin_expect_code = 'int main() { __builtin_expect(0,0); return 0; }'
+has_builtin_expect = cc.compiles(
+    has_builtin_expect_code,
+    name: 'has __builtin_expect',
+)
+has_attribute_code = '''
+  int f(void) __attribute__ (());
+  int main() { return 0; }
+'''
+has_attribute = cc.compiles(has_attribute_code, name: 'has __attribute__')
+
+cdata = configuration_data(
+    {
+        'HAVE_STDBOOL_H': has_stdbool_header,
+        'HAVE___BUILTIN_EXPECT': has_builtin_expect,
+        'HAVE___ATTRIBUTE__': has_attribute,
+    },
+)
+
+configure_file(
+    format: 'cmake@',
+    input: 'src/config.h.in',
+    output: 'config.h',
+    configuration: cdata,
+)
+
+varr = meson.project_version().split('.')
+vdata = configuration_data(
+    {
+        'PROJECT_VERSION_MAJOR': varr[0],
+        'PROJECT_VERSION_MINOR': varr[1],
+        'PROJECT_VERSION_PATCH': varr[2],
+        'PROJECT_VERSION_GFM': varr[4],
+    },
+)
+
+configure_file(
+    format: 'cmake@',
+    input: 'src/cmark-gfm_version.h.in',
+    output: 'cmark-gfm_version.h',
+    configuration: vdata,
+)
+
+edata = configuration_data(
+    {
+        'CMARK_GFM_STATIC_DEFINE': get_option('default_library') == 'static',
+    },
+)
+
+configure_file(
+    input: 'cmark-gfm_export.h.meson',
+    output: 'cmark-gfm_export.h',
+    configuration: edata,
+)
+
+configure_file(
+    input: 'cmark-gfm-extensions_export.h.meson',
+    output: 'cmark-gfm-extensions_export.h',
+    configuration: edata,
+)
+
+src = files(
+    'extensions/autolink.c',
+    'extensions/core-extensions.c',
+    'extensions/ext_scanners.c',
+    'extensions/ext_scanners.h',
+    'extensions/strikethrough.c',
+    'extensions/table.c',
+    'extensions/tagfilter.c',
+    'extensions/tasklist.c',
+    'src/arena.c',
+    'src/blocks.c',
```

Comment:
Shouldn't these be defined as two different libraries?

Reply:
Yes upstream defines them as two separate libraries so I'll make a follow-up commit to match that.